plugins/processors/awsapplicationsignals/internal/prune/metric_pruner.go

```diff
@@ -21,6 +22,9 @@ func (p *MetricPruner) ShouldBeDropped(attributes pcommon.Map) (bool, error) {
 				return true, errors.New("Metric attribute " + attributeKey + " must contain only ASCII characters.")
 			}
 		}
+		if _, ok := attributes.Get(common.MetricAttributeTelemetrySource); !ok {
```

Comment:
are there any risks in dropping metrics without Telemetry.Source? Are there any edge cases where a customer might emit application signals metrics without this attribute?

Reply:
In the case where customer is using preview SDK, all the metrics will be dropped. Those SDKs are deprecated though. Also even the traffic is sent to logs, Application Signals has stopped processing those logs emitted from preview SDK while using the new CloudWatch agent.